src/caches/sdirk_caches.jl

```diff
@@ -50,6 +56,8 @@ DiffEqBase.@def iipnlcachefields begin
     jac_config = nothing
     linsolve = nothing
     z₊ = similar(z)
+    zs = [zero(vec(z)) for i in 1:alg.nlsolve.n+1]
+    gs = [zero(vec(z)) for i in 1:alg.nlsolve.n+1]
```

Comment:
The OOP version prehapes doesn’t need explicit initialization. [vec(z) for i in 1:alg.nlsolve.n+1]

Reply:
Okay I think I should deal with this in the coming PR.
But I couldn't get why? Wouldn't doing only vec would make all the elements of the array same?

Reply:
Because you are going to populate them in the OOP AA anyway.

Reply:
This is resolved in the latest commit.